tests/testthat/test-util.R

```diff
@@ -46,24 +46,27 @@ test_that("resolve_env", {
   set.seed(1)
   v <- paste(sample(c(LETTERS, 0:9, "_"), 20, replace = TRUE), collapse = "")
   vv <- paste0("$", v)
-  expect_identical(resolve_env(v), list(v))
-  expect_error(resolve_env(vv),
-               sprintf("Environment variable '%s' is not set", v))
-  expect_identical(withr::with_envvar(setNames("value", v), resolve_env(vv)),
-                   list("value"))
+  expect_identical(resolve_env(c(x = v), "loc"), list(x = v))
+  expect_error(
+    resolve_env(c(x = vv), "loc"),
+    sprintf("Environment variable '%s' is not set.*used in loc:x", v))
+  expect_identical(
+    withr::with_envvar(setNames("value", v), resolve_env(c(x = vv), "loc")),
+    list(x = "value"))
 })
 
 
 test_that("resolve_env skips non-scalars", {
   set.seed(1)
   v <- paste(sample(c(LETTERS, 0:9, "_"), 20, replace = TRUE), collapse = "")
   vv <- paste0("$", v)
-  expect_identical(resolve_env(v), list(v))
+  expect_identical(resolve_env(c(x = v)), list(x = v))
```

Comment:
Note mostly to self - so, `used_in` (no default value) is not specified when calling resolve_env here - but it should not be needed in the test. Hence the test passes, and if `used_in` were required at some point during resolve_env, the test would correctly fail.

Reply:
I think this is the only time you call resolve_env in such a way though, so looked like it needed exploring at first glance.

Reply:
thanks, I had thought I'd caught all of these